samples/backend/csharp/Controllers/ChatController.cs

```diff
@@ -19,6 +20,78 @@ public ChatController(ISemanticKernelApp semanticKernelApp)
         _semanticKernelApp = semanticKernelApp;
     }
 
+    private (int MessageIndex, int FileIndex, IFormFile File) GetPosition(IFormFile formFile)
+    {
+        const string pattern = @"messages\[(\d+)\]\.files\[(\d+)\]";
+
+        var match = Regex.Match(formFile.Name, pattern);
+        if (match.Success && int.TryParse(match.Groups[1].Value, out var messageIndex) && int.TryParse(match.Groups[2].Value, out var fileIndex))
+        {
+            return (messageIndex, fileIndex, formFile);
+        }
+        throw new ArgumentException("Malformed multipart request: Invalid file name.");
+
+    }
+
+    private async Task<AIChatRequest> RequestFromMultipart(IFormFileCollection formFiles)
+    {
+        using var jsonFileStream = formFiles
+            .First(f => f.Name == "json")
+            .OpenReadStream();
+        if (jsonFileStream is null)
+        {
+            throw new Exception("Malformed multipart request: Missing json part.");
+        }
+        var request = await JsonSerializer.DeserializeAsync<AIChatRequest>(jsonFileStream);
+        if (request is null)
+        {
+            throw new Exception("Malformed multipart request: Invalid json part.");
+        }
+        foreach (var (messageIndex, fileIndex, file) in formFiles.Where(f => f.Name != "json").Select(GetPosition).OrderBy(p => p.MessageIndex).ThenBy(p => p.FileIndex))
+        {
+            using var fileStream = file.OpenReadStream();
+            if (request.Messages.Count <= messageIndex)
+            {
+                throw new Exception("Malformed multipart request: Invalid message index.");
+            }
+            var message = request.Messages[messageIndex];
+            message.Files ??= new List<AIChatFile>();
+            if (message.Files.Count != fileIndex)
+            {
+                throw new Exception("Malformed multipart request: Invalid file index.");
+            }
+            var fileData = await BinaryData.FromStreamAsync(fileStream);
+            message.Files.Add(new AIChatFile
+            {
+                ContentType = file.ContentType,
+                Data = fileData
+            });
+        }
+        return request;
+    }
+
+
+    [HttpPost]
+    [Consumes("multipart/form-data")]
+    public async Task<IActionResult> ProcessMessage(IFormFileCollection files)
+    {
+        try
+        {
+            var request = await RequestFromMultipart(files);
+            var session = request.SessionState switch
+            {
+                Guid sessionId => await _semanticKernelApp.GetSession(sessionId),
+                _ => await _semanticKernelApp.CreateSession(Guid.NewGuid())
```

Comment:
Let's not call this local or the type "semantic kernel app". SK is just the library being used to implement the functionality. It should be named based on what it does rather than on how it's implemented.

Reply:
will work with @rohit-ganguly to improve naming for our samples (in a different PR)